core/src/replay_stage.rs

```diff
@@ -2034,11 +2045,17 @@ impl ReplayStage {
 
         trace!("{} reached_leader_slot", my_pubkey);
 
-        let parent = bank_forks
-            .read()
-            .unwrap()
-            .get(parent_slot)
-            .expect("parent_slot doesn't exist in bank forks");
+        let parent = bank_forks.read().unwrap().get(parent_slot);
```

Comment:
We can let else:
```rust
let Some(parent) = bank_forks.read().unwrap().get(parent_slot) else {
    warn!("Poh recorder parent slot {parent_slot} is missing from bank_forks. This indicates that we are in the middle of a dump and repair. Unable to start leader");
    return false;
}
```

Reply:
Done.

Reply:
can you  update the warn! message to match what I wrote above?

Reply:
Done.